W-14075162 Registration certificate renewal docs (#383)
mulesoft/docs-gateway: gateway/1.5/modules/ROOT/pages/flex-gateway-renew-certificate.adoc

= Renewing Flex Gateway Registration
ifndef::env-site,env-github[]
include::_attributes.adoc[]
endif::[]

Renewing Flex Gateway registration ensures the continued operation of your applications. You renew registration by using the `flexctl` CLI.

> To avoid potential disruptions, verify the success of the renewal process by using the `flexctl inspect` command.

Reviewer: `registration inspect`

* <<renewal-cli-overview,Registration Renewal CLI Command Overview>>
* <<inspection-cli-overview,Registration Inspection CLI Command Overview>>

== Before You Begin

* xref:flex-install.adoc[Download Flex Gateway].
+
To invoke the registration renewal `flexctl` command, download the latest version of Flex Gateway. It is not necessary to register and run this version of the gateway.

[[renewal-cli-overview]]
== Registration Renewal CLI Command Overview

Flex Gateway includes the following CLI command to renew registration:

----
flexctl registration renew
----

The command renews an existing Flex Gateway registration with Runtime Manager. For authentication, you specify *either* user credentials or connected appl credentials. Authentication with user credentials requires the `--username` and `--password` flags. Authentication with connected app credentials requires the `--client-id` and `--client-secret` flags.

Invoke the following `help` command for information about usage:

[source,ssh]
----
flexctl registration renew --help
----

For Flex Gateway running in a container, invoke `help` as part of the container `run` command:

[tabs]
====
Docker::
+
[source,ssh]
----

> docker run --entrypoint flexctl mulesoft/flex-gateway renew --help

Reviewer: `docker run --entrypoint flexctl mulesoft/flex-gateway registration renew --help`

----

Podman::
+
[source,ssh]
----

> podman run --entrypoint flexctl docker.io/mulesoft/flex-gateway renew --help

Reviewer: `podman run --entrypoint flexctl docker.io/mulesoft/flex-gateway registration renew --help`

----
====

Refer to the following topics for more information about renewal command usage:

* <<renew-with-user-password,Renew Registration with User Credentials>>
* <<renew-with-conn-app,Renew Registration with Connected App Credentials>>

[[renew-with-user-password]]
== Renew Registration with User Credentials

Flex Gateway includes the following CLI command to renew registration using Anypoint Platform user credentials:

[source,ssh]
----
flexctl registration renew --username=<your-username> --password=<your-password> \
 <path-to-registration-file>
----

[NOTE]
====
The user must include _Read Servers_ and _Manage Servers_ permissions for Runtime Manager.
====

> Replace `<path-to-registration-file>` with the path and filename of the existing `registration.yaml` file. Typically, the existing registration file is overwritten by the renewal process. You can optionally output the new registration file to a directory by using the `--output-directory` flag.

Reviewer: the existing registration file is overwritten only if the output directory is the same one as the directory where the registration file is

For Flex Gateway running in a container, invoke `renew` as part of the container `run` command:

[tabs]
====
Docker::
+
[source,ssh]
----
docker run --entrypoint flexctl \
-v "$(pwd)":/renew \
-v <path-to-registration-directory>:/registration \
-u $UID mulesoft/flex-gateway \

> renew \

Reviewer: `registration renew`

--username=<your-username> \
--password=<your-password> \
--output-directory=/renew \
/registration/<registration-filename>
----

Podman::
+
[source,ssh]
----
podman run --entrypoint flexctl --userns=keep-id \
-v "$(pwd)":/renew:Z \
-v <path-to-registration-directory>:/registration:Z \
-u $UID docker.io/mulesoft/flex-gateway \

> renew \

Reviewer: `registration renew`

[source,ssh]
----
flexctl registration renew --client-id=<your-client-id> --client-secret=<your-client-secret> \
 <path-to-registration-file>
----

If you do not have a username or password for Anypoint Platform, you can configure connected apps via Anypoint Access Management:

* xref:access-management::connected-apps-developers.adoc#create-a-connected-app[Configure a Connected App]
** Include the following scopes:
*** Read Servers
*** Manage Servers
*** View Organization
** Save the *Id* and *Secret* of the Connected app you configure, then use these credentials in the `flexctl registration renew` command.

Replace `<path-to-registration-file>` with the path and filename of the existing `registration.yaml` file. Typically, the existing registration file is overwritten by the renewal process. You can optionally output the new registration file to a directory by using the `--output-directory` flag.

For Flex Gateway running in a container, invoke `renew` as part of the container `run` command:

[tabs]
====
Docker::
+
[source,ssh]
----
docker run --entrypoint flexctl \
-v "$(pwd)":/renew \
-v <path-to-registration-directory>:/registration \
-u $UID mulesoft/flex-gateway \

> renew \

Reviewer: `registration renew`

--client-id=<your-client-id> \
--client-secret=<your-client-secret> \
--output-directory=/renew \
/registration/<registration-filename>
----

Podman::
+
[source,ssh]
----
podman run --entrypoint flexctl --userns=keep-id \
-v "$(pwd)":/renew:Z \
-v <path-to-registration-directory>:/registration:Z \
-u $UID docker.io/mulesoft/flex-gateway \

> renew \

Reviewer: registration renew

--client-id=<your-client-id> \
--client-secret=<your-client-secret> \
--output-directory=/renew \
/registration/<registration-filename>
----
====

Invoking the command results in the following output:

[source,text]
----
Starting registration renewal, please be patient.
Registration renewal completed, the configuration files were written in directory ".".
For security, modify the file permissions to restrict production scenario access to the user running flex.

Side effect: the following files were written in the current directory
registration.yaml
----

[[inspection-cli-overview]]
== Registration Inspection CLI Command Overview

Flex Gateway includes the following CLI command to inspect registration:

----
flexctl registration inspect
----

The command inspects your Flex Gateway registration, extracting useful information you use to verify whether the renewal command was successful.

Invoke the following `help` command for information about usage:

[source,ssh]
----
flexctl registration inspect --help
----

For Flex Gateway running in a container, invoke `help` as part of the container `run` command:

[tabs]
====
Docker::
+
[source,ssh]
----

> docker run --entrypoint flexctl mulesoft/flex-gateway inspect --help

Reviewer: `docker run --entrypoint flexctl mulesoft/flex-gateway registration inspect --help`

----

Podman::
+
[source,ssh]
----

> podman run --entrypoint flexctl docker.io/mulesoft/flex-gateway inspect --help

Reviewer: `podman run --entrypoint flexctl docker.io/mulesoft/flex-gateway registration inspect --help`

----
====

Refer to the following topics for more information about inspection command usage:

* <<inspect-with-custom-url,Inspect Registration Status with a Custom URL>>
* <<inspect-with-reg-file,Inspect Registration Status with a Registration File>>

[[inspect-with-custom-url]]
== Inspect Registration Status with a Custom URL

> You can inspect a registration by specifying the base URL of the Flex Gateway Admin API. The default value is `http://localhost:9999`. Omit the `--flex-gateway-admin-api-url` flag if using the default value.

Reviewer: I would also clarify that if no flags are passed, by default it will look up on the flex gateway admin api

[source,ssh]
----
flexctl registration inspect \
--flex-gateway-admin-api-url=<your-custom-admin-api-url>
----

For Flex Gateway running in a container, invoke `inspect` as part of the container `exec` command. Replace `<container-name-or-id>` with the running container reference:

[tabs]
====
Docker::
+
[source,ssh]
----

> docker exec -u $UID <container-name-or-id> flexctl inspect \

Reviewer: `flexctl registration inspect`

--flex-gateway-admin-api-url=<your-custom-admin-api-url>
----

Podman::
+
[source,ssh]
----

> podman exec -u $UID <container-name-or-id> flexctl inspect \

Reviewer: `flexctl registration inspect`

--flex-gateway-admin-api-url=<your-custom-admin-api-url>
----
====

Invoking the command results in the following example output:

[source,text]
----
{“expiration_date”: “2024-01-16 16:20:30 +0000 UTC”}
----

[[inspect-with-reg-file]]
== Inspect Registration Status with a Registration File

You can inspect a registration by extracting the information from your `registration.yaml` file. Specify the path and filename.

[source,ssh]
----
flexctl registration inspect --file=<path-to-registration-file>
----

For Flex Gateway running in a container, invoke `inspect` as part of the container `run` command:

[tabs]
====
Docker::
+
[source,ssh]
----
docker run --entrypoint flexctl \
-v <path-to-registration-directory>:/registration \
-u $UID mulesoft/flex-gateway \

> inspect --file=/registration/<registration-filename>

Reviewer: `registration inspect`

----

Podman::
+
[source,ssh]
----
podman run --entrypoint flexctl --userns=keep-id \
-v <path-to-registration-directory>:/registration:Z \
-u $UID docker.io/mulesoft/flex-gateway \

> inspect --file=/registration/<registration-filename>

Reviewer: `registration inspect`

Attempting to run Flex Gateway with an expired certificate in Connected Mode results in the following:

* New replicas fail to download API configurations from Anypoint Platform, thereby becoming unusable. Existing clusters fail to reload.
* Logs and metrics fail to upload to Anypoint Platform. Troubleshooting using Anypoint Platform is not possible.
* Metering information fails to upload to Anypoint Platform. MuleSoft is unable to collect or report usage metrics.

Attempting to run Flex Gateway with an expired certificate in Local Mode results in the following:

* Metering information fails to upload to Anypoint Platform. MuleSoft is unable to collect or report usage metrics.

If a certificate expires, you can still invoke `flexctl registration renew`, which updates the expired certificate.

To avoid disruptions:

. <<before-you-begin,Download the latest version of Flex Gateway.>>
. <<renewal-cli-overview,Renew your Flex Gateway registration.>>
. <<inspection-cli-overview,Verify (inspect) the success of the renewal process.>>

[[before-you-begin]]
== Before You Begin

* xref:flex-install.adoc[Download Flex Gateway].
+

> To invoke the registration renewal `flexctl` command, download the latest version of Flex Gateway. It is not necessary to register and run this version of the gateway.

Reviewer 2: ```suggestion
To invoke the registration renewal `flexctl` command, download the latest version of Flex Gateway. You don't need to register and run this version of the gateway.
```

Author: Fixed!

[[renewal-cli-overview]]

> == Registration Renewal

Reviewer 2: What about making these tasks?

Renew Your Registration

. Use the following CLI command to new registration:
<command>
. When prompted to authenticate, use either ...

Author: I made "Renew Your Registration" into a task step, along with "Download Flex Gateway", and "Verify Your Registration." Each one of these tasks now has subtasks.

Flex Gateway includes the following CLI command to renew registration:

----
flexctl registration renew
----

The command renews an existing Flex Gateway registration with Runtime Manager. For authentication, you specify *either* user credentials or connected appl credentials. Authentication with user credentials requires the `--username` and `--password` flags. Authentication with connected app credentials requires the `--client-id` and `--client-secret` flags.

> Invoke the following `help` command for information about usage:

Reviewer 2: Maybe this is a separate task?

Author: I'm thinking that the "help" command isn't actually part of the task. It's more of a reference for something they can do, as opposed to something they need to do to renew their registration.

Reviewer 2: What about moving the help bits to a separate topic at the end?

Author: Fixed!

[source,ssh]
----
flexctl registration renew --help
----

For Flex Gateway running in a container, invoke `help` as part of the container `run` command:

[tabs]
====
Docker::
+
[source,ssh]
----
docker run --entrypoint flexctl mulesoft/flex-gateway registration renew --help
----

Podman::
+
[source,ssh]
----
podman run --entrypoint flexctl docker.io/mulesoft/flex-gateway registration renew --help
----
====

Refer to the following topics for more information about renewal command usage:

* <<renew-with-user-password,Renew Registration with User Credentials>>
* <<renew-with-conn-app,Renew Registration with Connected App Credentials>>

[[renew-with-user-password]]

> === Renew Registration with User Credentials

Reviewer 2: Rephrase as a task?

Author: This is one of two methods they can use to renew registration, so it in itself shouldn't be a task step. However, I made subtasks for each of these methods.

Reviewer 2: I disagree. 

I think you can simplify and flatten this structure significantly.

In the intro, introduce the two methods and have the bullet list with links to each (as you do later).
Talk about when you would use one vs the other.

== Before You Begin
installation stuff

== Renew Registration with User Credentials
. Step 1
. Step 2

== Renew Registration with Connected App Credentials
. Step 1
. Step 2

== Get Help on Renew CLI Commands

Author: Fixed! I flattened the structure as suggested.

> Flex Gateway includes the following CLI command to renew registration using Anypoint Platform user credentials:

Reviewer 2: To..., use this...:

Author: This text was removed in the whole task-ification effort.

[source,ssh]
----
flexctl registration renew --username=<your-username> --password=<your-password> \
 <path-to-registration-file>
----

> Replace `<path-to-registration-file>` with the path and filename of the existing `registration.yaml` file. You can optionally output the new registration file to a directory by using the `--output-directory` flag. The `renew` command overwrites the existing registration file if you specify the file's directory as the output directory.

Reviewer 2: This would be a paragraph under the step.

Author: Fixed. I made it into a step.

> IMPORTANT: Create a new directory called `flex-renew-registration` (or similar) and then run the `renew` command in this new directory. The command creates registration files in this location.

Reviewer 2: Make this the first step in the task. It's kind of a prereq.

Author: Fixed!

[tabs]
====
Docker::
+
[source,ssh]
----
docker run --entrypoint flexctl \
-v "$(pwd)":/renew \
-v <path-to-registration-directory>:/registration \
-u $UID mulesoft/flex-gateway \
registration renew \
--username=<your-username> \
--password=<your-password> \
--output-directory=/renew \
/registration/<registration-filename>
----

Podman::
+
[source,ssh]
----
podman run --entrypoint flexctl --userns=keep-id \
-v "$(pwd)":/renew:Z \
-v <path-to-registration-directory>:/registration:Z \
-u $UID docker.io/mulesoft/flex-gateway \
registration renew \
--username=<your-username> \
--password=<your-password> \
--output-directory=/renew \
/registration/<registration-filename>
----
====

Invoking the command results in the following output:

[source,text]
----
Starting registration renewal, please be patient.
Registration renewal completed, the configuration files were written in directory ".".
For security, modify the file permissions to restrict production scenario access to the user running flex.

Side effect: the following files were written in the current directory
registration.yaml
----

[[renew-with-conn-app]]

> === Renew Registration with Connected App Credentials

Reviewer 2: task

Author: This is one of two methods they can use to renew registration, so it in itself shouldn't be a task step. However, I made subtasks for each of these methods.

Author: Fixed!

> Flex Gateway includes the following CLI command to renew registration using connected app credentials:

Reviewer 2: To..., do...

Author: This text was removed in the whole task-ification effort.

[source,ssh]
----
flexctl registration renew --client-id=<your-client-id> --client-secret=<your-client-secret> \
 <path-to-registration-file>
----

> If you do not have a username or password for Anypoint Platform, you can configure connected apps via Anypoint Access Management:

Reviewer 2: ```suggestion
If you don't have a username or password for Anypoint Platform, you can configure connected apps via Anypoint Access Management:
```

Author: Fixed!

> * xref:access-management::connected-apps-developers.adoc#create-a-connected-app[Configure a Connected App]

Reviewer 2: Should this be an ordered step? (Step 1)

Author: Fixed!

> ** Include the following scopes:

Reviewer 2: Step a.

Author: Fixed!

*** Read Servers
*** Manage Servers
*** View Organization

> ** Save the *Id* and *Secret* of the Connected app you configure, then use these credentials in the `flexctl registration renew` command.

Reviewer 2: Step b.

Author: Fixed!

> ** Save the *Id* and *Secret* of the Connected app you configure, then use these credentials in the `flexctl registration renew` command.

Reviewer 2: ```suggestion
** Save the *Id* and *Secret* of the Connected App you configure, then use these credentials in the `flexctl registration renew` command.
```

Reviewer 2: FWIW, I only used Connected App (sentence case) when referring to the feature. When referring to the thing - my connected app, I used lc.

Author: Fixed!

> Replace `<path-to-registration-file>` with the path and filename of the existing `registration.yaml` file. You can optionally output the new registration file to a directory by using the `--output-directory` flag. The `renew` command overwrites the existing registration file if you specify the file's directory as the output directory.

Reviewer 2: Paragraph under the step.

Author: Fixed. I made it into a step.

> IMPORTANT: Create a new directory called `flex-renew-registration` (or similar) and then run the `renew` command in this new directory. The command creates registration files in this location.

Reviewer 2: First step. Prereq.

Author: Fixed!

[tabs]
====
Docker::
+
[source,ssh]
----
docker run --entrypoint flexctl \
-v "$(pwd)":/renew \
-v <path-to-registration-directory>:/registration \
-u $UID mulesoft/flex-gateway \
registration renew \
--client-id=<your-client-id> \
--client-secret=<your-client-secret> \
--output-directory=/renew \
/registration/<registration-filename>
----

Podman::
+
[source,ssh]
----
podman run --entrypoint flexctl --userns=keep-id \
-v "$(pwd)":/renew:Z \
-v <path-to-registration-directory>:/registration:Z \
-u $UID docker.io/mulesoft/flex-gateway \
registration renew \
--client-id=<your-client-id> \
--client-secret=<your-client-secret> \
--output-directory=/renew \
/registration/<registration-filename>
----
====

Invoking the command results in the following output:

[source,text]
----
Starting registration renewal, please be patient.
Registration renewal completed, the configuration files were written in directory ".".
For security, modify the file permissions to restrict production scenario access to the user running flex.

Side effect: the following files were written in the current directory
registration.yaml
----

[[inspection-cli-overview]]

> == Registration Inspection

Reviewer 2: Task

== Inspect Registration

Author: Fixed!

> Flex Gateway includes the following CLI command to inspect registration:

Reviewer 2: . Run the following command to...

Author: Fixed!

----
flexctl registration inspect
----

The command inspects your Flex Gateway registration, extracting information you use to verify whether the renewal command was successful.

> Invoke the following `help` command for information about usage:

Reviewer 2: New task?

Author: See note above about the "help" command not being a part of the overall steps.

Author: Fixed! I moved the help instructions to a level 2 section at the bottom.

[source,ssh]
----
flexctl registration inspect --help
----

For Flex Gateway running in a container, invoke `help` as part of the container `run` command:

[tabs]
====
Docker::
+
[source,ssh]
----
docker run --entrypoint flexctl mulesoft/flex-gateway registration inspect --help
----

Podman::
+
[source,ssh]
----
podman run --entrypoint flexctl docker.io/mulesoft/flex-gateway registration inspect --help
----
====

Refer to the following topics for more information about inspection command usage:

* <<inspect-with-custom-url,Inspect Registration Status with an Admin API URL>>
* <<inspect-with-reg-file,Inspect Registration Status with a Registration File>>

[NOTE]
====
If you call `flexctl registration inspect` without flags, the command inspects registration status by using the default Admin API URL.
====

[[inspect-with-custom-url]]

> === Inspect Registration Status with an Admin API URL

Reviewer 2: This is more of a task

Reviewer 2: All the other sections are worded as task (as expected).

Author: Fixed!

----
====

Invoking the command results in the following example output:

[source,text]
----
{“expiration_date”: “2024-01-16 16:20:30 +0000 UTC”}
----

[[inspect-with-reg-file]]
=== Inspect Registration Status with a Registration File

You can inspect a registration by extracting the information from your `registration.yaml` file. Specify the path and filename.

[source,ssh]
----
flexctl registration inspect --file=<path-to-registration-file>
----

For Flex Gateway running in a container, invoke `inspect` as part of the container `run` command:

[tabs]
====
Docker::
+
[source,ssh]
----
docker run --entrypoint flexctl \
-v <path-to-registration-directory>:/registration \
-u $UID mulesoft/flex-gateway \
registration inspect --file=/registration/<registration-filename>
----

Podman::
+
[source,ssh]
----
podman run --entrypoint flexctl --userns=keep-id \
-v <path-to-registration-directory>:/registration:Z \
-u $UID docker.io/mulesoft/flex-gateway \
registration inspect --file=/registration/<registration-filename>
----
====

Invoking the command results in the following example output:

[source,text]
----
{“expiration_date”: “2024-01-16 16:20:30 +0000 UTC”}
----

== See Also

> * xref:flex-install.adoc[Download Flex Gateway]

Reviewer 2: If you omit the the hardcoded text, it will propagate automatically and be easier to maintain.

Author: Fixed!

. Use the following command:
+
[tabs]
====
Linux::
+
[source,ssh]
----
flexctl registration renew --username=<your-username> --password=<your-password> \
 <path-to-registration-file>
----
+
Use `sudo` if you encounter file permission issues when running this command.

Docker::
+
For Flex Gateway running in a container, invoke `renew` as part of the container `run` command.
+
[source,ssh]
----
docker run --entrypoint flexctl \
-v "$(pwd)":/renew \
-v <path-to-registration-directory>:/registration \
-u $UID mulesoft/flex-gateway \
registration renew \
--username=<your-username> \
--password=<your-password> \
--output-directory=/renew \
<path-to-registration-file>
----

Podman::
+
For Flex Gateway running in a container, invoke `renew` as part of the container `run` command.
+
[source,ssh]
----
podman run --entrypoint flexctl --userns=keep-id \
-v "$(pwd)":/renew:Z \
-v <path-to-registration-directory>:/registration:Z \
-u $UID docker.io/mulesoft/flex-gateway \
registration renew \
--username=<your-username> \
--password=<your-password> \
--output-directory=/renew \
<path-to-registration-file>
----
====

> . Replace `<path-to-registration-file>` with the path and filename of the existing `registration.yaml` file. You can optionally output the new registration file to a directory by using the `--output-directory` flag. The `renew` command overwrites the existing registration file if you specify the file's directory as the output directory.

Reviewer: a small change, the renew command cannot overwrite an existing registration

Author: Fixed!

-u $UID mulesoft/flex-gateway \
registration renew \
--client-id=<your-client-id> \
--client-secret=<your-client-secret> \
--output-directory=/renew \
<path-to-registration-file>
----

Podman::
+
For Flex Gateway running in a container, invoke `renew` as part of the container `run` command.
+
[source,ssh]
----
podman run --entrypoint flexctl --userns=keep-id \
-v "$(pwd)":/renew:Z \
-v <path-to-registration-directory>:/registration:Z \
-u $UID docker.io/mulesoft/flex-gateway \
registration renew \
--client-id=<your-client-id> \
--client-secret=<your-client-secret> \
--output-directory=/renew \
<path-to-registration-file>
----
====

. Replace `<path-to-registration-file>` with the path and filename of the existing `registration.yaml` file. You can optionally output the new registration file to a directory by using the `--output-directory` flag. The `renew` command overwrites the existing registration file if you specify the file's directory as the output directory.

. Replace <your-client-id> and <your-client-secret> with the connected app credentials.

. Run the command.
+
The output should include the message:
+
[source,text]
----

> Registration renewal completed

Reviewer: this is how the final message looks like

```
Starting registration renewal, please be patient.
Registration renewal completed, new expiration date is 2025-09-25 19:27:32 +0000 UTC. Configuration files were written in directory "./resources/examples/common/config.d". For security, modify the file permissions to restrict production scenario access to the user running flex.
```

Attempting to run Flex Gateway with an expired certificate in Connected Mode results in the following:

* New replicas fail to download API configurations from Anypoint Platform, thereby becoming unusable. Existing clusters fail to reload.
* Logs and metrics fail to upload to Anypoint Platform. Troubleshooting using Anypoint Platform is not possible.
* Metering information fails to upload to Anypoint Platform. MuleSoft is unable to collect or report usage metrics.

Attempting to run Flex Gateway with an expired certificate in Local Mode results in the following:

* Metering information fails to upload to Anypoint Platform. MuleSoft is unable to collect or report usage metrics.

If a certificate expires, you can still invoke `flexctl registration renew`, which updates the expired certificate.

To avoid disruptions:

. <<before-you-begin,Download the latest version of Flex Gateway.>>
. <<renewal-cli-overview,Renew your Flex Gateway registration.>>
. <<inspection-cli-overview,Verify the success of the renewal process.>>

[[before-you-begin]]

> === 1. Download Latest Version of Flex Gateway

Reviewer 2: We don't put numbers in headings unless it's a complicated Tutorial.

Reviewer 2: Maybe this should just be step 1.
Usually we have a Before You Begin topic, but this is something they actually have to do.

Author: Fixed!

To invoke the registration renewal `flexctl` command, download the latest version of Flex Gateway. You don't need to register and run this version of the gateway.

* xref:flex-install.adoc[]

[[renewal-cli-overview]]

> === 2. Renew Your Registration

Reviewer 2: Just make this one == H2 heading - "Renew Your Registration" and put the installation as step 1 of the task.

> Flex Gateway includes the following CLI command to renew registration:

Reviewer 2: This is not worded as a step in a task.

This is what I mean:

To renew registration: 

. Download the latest....
. Enter the following command:

Author: Fixed!

----
flexctl registration renew
----

> The command renews an existing Flex Gateway registration with Runtime Manager. For authentication, you run the command with either user credentials or with connected app credentials. Authentication with user credentials requires the `--username` and `--password` flags. Authentication with connected app credentials requires the `--client-id` and `--client-secret` flags.

Reviewer 2: This is a paragraph under step 2.

Author: Fixed! I moved this contextual information to the page's introduction.

> You can run the following `help` command for information about usage:

Reviewer 2: I think the help instructions interrupt the flow.

Author: Fixed! I moved the help instructions to a level 2 section at the bottom.

[tabs]
====
Linux::
+
[source,ssh]
----
flexctl registration renew --help
----
+
Use `sudo` if you encounter file permission issues when running this command.

Docker::
+
For Flex Gateway running in a container, invoke `help` as part of the container `run` command.
+
[source,ssh]
----
docker run --entrypoint flexctl mulesoft/flex-gateway registration renew --help
----

Podman::
+
For Flex Gateway running in a container, invoke `help` as part of the container `run` command.
+
[source,ssh]
----
podman run --entrypoint flexctl docker.io/mulesoft/flex-gateway registration renew --help
----
====

> The `flexctl` CLI provides two methods to renew your registration:

Reviewer 2: Why is this here? Shouldn't this be in the intro?

OK, I see what you did. So, I think the first procedure (overview) duplicates what's below, which is confusing.

Author: Fixed!

* <<renew-with-user-password,Method 1: Renew Registration with Anypoint Platform User Credentials>>
* <<renew-with-conn-app,Method 2: Renew Registration with Connected App Credentials>>

[[renew-with-user-password]]

> ==== Method 1: Renew Registration with Anypoint Platform User Credentials

Reviewer 2: You don't need the Method 1 here.

Just move the intro to the beginning and discuss there. It's obvious that these are two alternatives.

Reviewer 2: The problem with this structure is that you have 4 levels of hierarchy.

Author: Fixed!
